confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -18,133 +18,258 @@
+        return conf
 
-        def GET(self, url, decoder=None):
-            return self.send_request(url, method='GET', decoder=decoder)
+    def close(self):
+        self.session.close()
 
-        def POST(self, url, body, encoder=None, decoder=None, **kwargs):
-            return self.send_request(url, method='POST',
-                                     body=body, encoder=encoder, decoder=decoder,
-                                     **kwargs)
+    def GET(self, url):
+        return self.send_request(url, method='GET')
 
-        def DELETE(self, url):
-            return self.send_request(url, method='DELETE')
+    def POST(self, url, body, **kwargs):
 
-        def PUT(self, url, body=None, encoder=None, decoder=None):
-            return self.send_request(url, method='PUT', body=body,
-                                     encoder=encoder, decoder=decoder)
+        return self.send_request(url, method='POST',
+                                 body=body)
 
-        def send_request(self, url, method, body=None,
-                         encoder=None, decoder=None, **kwargs):
+    def DELETE(self, url):
+        return self.send_request(url, method='DELETE')
 
-            _headers = self.default_headers
+    def PUT(self, url, body=None):
+        return self.send_request(url, method='PUT', body=body)
 
-            if body is not None:
-                body = json.dumps(body, cls=encoder)
-                _headers = {'Content-Length': str(len(body)),
-                            'Content-Type': "application/vnd.schemaregistry.v1+json"}
+    def send_request(self, url, method, body=None):
 
-            response = self.session.request(
-                method, url="/".join([self.base_url, url]),
-                headers=_headers, data=body)
+        _headers = self.default_headers
 
-            try:
-                if 200 <= response.status_code <= 299:
-                    return response.json(cls=decoder, **kwargs)
-                raise SchemaRegistryClientError(response.status_code,
-                                                response.json().get('message'))
-            except ValueError:
-                raise SchemaRegistryClientError(response.status_code,
-                                                response.content)
+        if body is not None:
+            body = json.dumps(body, cls=self.JSONDuckEncoder)
+            _headers = {'Content-Length': str(len(body)),
+                        'Content-Type': "application/vnd.schemaregistry.v1+json"}
 
-    def register_schema(self, subject_name, schema):
+        response = self.session.request(
+            method, url="/".join([self.base_url, url]),
+            headers=_headers, data=body)
+
+        try:
+            if 200 <= response.status_code <= 299:
+                return response.json()
+            raise SchemaRegistryError(response.json().get('message', None),
+                                      response.json().get('error_code', None))
```

Comment:
should also include HTTP status code in SchemaRegistryError in addition to the SR error code.